cli/src/pcluster/aws/ec2.py

```diff
@@ -244,9 +244,13 @@ def get_instance_type_info(self, instance_type):
     def get_supported_architectures(self, instance_type):
         """Return a list of architectures supported for the given instance type."""
         instance_info = self.get_instance_type_info(instance_type)
-        supported_architectures = instance_info.supported_architecture()
+        return self.extract_supported_architectures(instance_info)
 
-        # Some instance types support multiple architectures (x86_64 and i386). Filter unsupported ones.
+    @staticmethod
+    def extract_supported_architectures(instance_type_info):
+        """Return a list that filters out unsupported architectures that an instance type may have."""
+        supported_architectures = instance_type_info.supported_architecture()
+        # Some instance types support multiple architectures (x86_64 and i386). Filter out unsupported ones.
         return list(set(supported_architectures) & set(SUPPORTED_ARCHITECTURES))
```

Comment:
isn't this filtering out already done in InstanceTypeInfo.supported_architecture()?